Edit fruit trees, delete fruit trees, add user_id to schema

stefan-apr opened the pull request:
Build out edit functionality and modal component,
add edit and delete buttons to sidebar,
add user_id to fruit_tree_location schema,
make user_id and fruit_id into FKs,
update add to include user_id,
conditionally render edit/delete on sidebar by user_id,
build out delete functionality and modal component

thomatha reviewed (commented):
I've tested this locally, and found these issues:

EditModal:

- EditModal needs to remember fruit type, Edit button is disabled
- Updating location does not work, should be able to click on the map to change location, may not need radio buttons

DeleteModal:

- After delete, reload fruit locations on the map

stefan-apr commented:
Updated PR:
Updated edit modal to remember fruit type and populate fruit type selector with initial fruit type
Updated edit modal to remove radio buttons. Map now defaults to old tree location, and location can be adjusted by clicking somewhere
Handled the case of untouched edit modal elements by defaulting to old tree values.
Fixed an issue with the sidebar so it no longer appears before the selected fruit tree's data loads in.

Note: I currently don't know why the delete modal does not reload the fruit locations on the map after deleting. It makes a call to updateBounds(), which is exactly the same thing the add modal does. But that does not consistently work. In fact, this does not always work for the add modal either, though it usually does. I think it must be some wonkiness with state being asynchronously set, with no control over whether the bounds or the add/delete/edit finish first. Investigating it further.

thomatha commented on src/app/api/fruit_locations/route.ts:
```diff
@@ -237,17 +248,17 @@ export async function GET(request: Request) {
     const data = await request.json();
 
-    if(!data || !data.name || !data.latitude || !data.longitude || !data.fruit_id) {
+    if(!data || !data.name || !data.latitude || !data.longitude || !data.fruit_id || !data.user_id) {
```
It would be more secure to get the `user_id` from the user's server side session.  Allowing it to be passed in lets people impersonate others.

thomatha reviewed (commented):
Thanks, some comments

thomatha reviewed (approved):
Thank you for the changes!
